client/src/components/Donation/stripe-card-form.tsx

```diff
@@ -9,19 +9,45 @@ import {
 } from '@stripe/react-stripe-js';
 import { loadStripe } from '@stripe/stripe-js';
 import type {
-  Token,
   StripeCardNumberElementChangeEvent,
-  StripeCardExpiryElementChangeEvent
+  StripeCardExpiryElementChangeEvent,
+  PaymentIntent,
+  StripeError
 } from '@stripe/stripe-js';
 import React, { useState } from 'react';
 
 import envData from '../../../../config/env.json';
 import { AddDonationData } from './PaypalButton';
 
 const { stripePublicKey }: { stripePublicKey: string | null } = envData;
+
+export type handleAuthenticationType = (
+  clientSecret: string,
+  paymentMethod: string
+) =>
+  | Promise<
+      | {
+          paymentIntent: PaymentIntent;
+          error?: undefined;
+        }
+      | {
+          paymentIntent?: undefined;
+          error: StripeError;
+        }
+      | {
+          error: {
+            type: string;
+          };
+        }
+    >
+  | { error: { type: 'StripeNotLoaded' } };
+
 interface FormPropTypes {
   onDonationStateChange: (donationState: AddDonationData) => void;
-  postStripeCardDonation: (token: Token) => void;
+  postStripeCardDonation: (
+    paymentMethodId: string,
+    handleAuthentication: handleAuthenticationType
```

Comment:
```suggestion
    handleAuthentication: HandleAuthentication
```
I'll get around to adding a lint rule one of these days...

Reply:
The Lint rule will become annoying with externally typed declarations. E.g. the editor (monaco)

Reply:
It shouldn't apply to them - we don't lint packages, only our code.

Reply:
But, we import the types.

Reply:
For the editor, I cannot think of anything `SoAndSoType`, but you also have an aversion to `ISoAndSo`

Reply:
> But, we import the types.

Doesn't matter.  The linter would only check our definitions, not the imported ones.

Reply:
Ah, well then ignore me.